### PR TITLE
Pipeline-ify configuration/themes/upload APIs

### PR DESCRIPTION
**Pull request by halfdan**
refs #5508

### CODE REVIEW

**Comment by ErisDS** [comment]
API options are now merged, so this will need a bit of an update :wink:

**Comment by halfdan** [comment]
@ErisDS Updated

File: core/server/api/themes.js [hunk 1]
```diff
@@ -21,41 +24,57 @@ themes = {
      * @returns {Promise(Themes)}
      */
     browse: function browse(options) {
-        options = options || {};
+        var tasks;
 
-        return canThis(options.context).browse.theme().then(function () {
+        function handlePermissions(options) {
+            return canThis(options.context).browse.theme().then(function () {
+                return Promise.resolve();
```
Comment (ErisDS) [inline core/server/api/themes.js]: should be `return options;`

File: core/server/api/themes.js [hunk 2]
```diff
@@ -21,41 +24,57 @@ themes = {
...
+            }, function () {
+                return Promise.reject(new errors.NoPermissionError('You do not have permission to browse themes.'));
```
Comment (ErisDS) [inline core/server/api/themes.js]: This one should be in the form `return errors.handleAPIError(error, 'message here');` I think

**Comment by ErisDS** [comment]
The build failure on this is genuine :wink:

**Comment by ErisDS** [comment]
@halfdan Can you get this one fixed up plz?

**Comment by ErisDS** [comment]
Any chance of a fixup here or should I close?

**Comment by ErisDS** [comment]
Gonna close for now so that other people can pick up these pieces.

**Pull request closed without merging**